poseidon2/io.nim

```diff
@@ -3,32 +3,35 @@ import constantine/math/arithmetic
 import constantine/math/io/io_bigints
 import constantine/math/config/curves
 
-func fromBytes*(_: type F, bytes: openArray[byte]): F =
+export curves
+
+func fromOpenArray(_: type F, bytes: openArray[byte]): F =
+  F.fromBig(B.unmarshal(bytes, littleEndian))
+
+func fromBytes*(_: type F, bytes: array[31, byte]): F =
   ## Converts bytes into a field element. The byte array is interpreted as a
-  ## canonical little-endian big integer. The array should be of length 31 bytes
-  ## or less to ensure that it fits in a field of 254 bits. The remaining
-  ## 6 most-significant bits are set to 0.
-  assert bytes.len <= 31
-  var padded: array[32, byte]
-  copyMem(addr padded[0], unsafeAddr bytes[0], bytes.len)
-  let bigint = B.unmarshal(padded, littleEndian)
-  return F.fromBig(bigint)
+  ## canonical little-endian big integer.
+  F.fromOpenArray(bytes)
+
+func toBytes*(element: F): array[32, byte] =
+  ## Converts a field element into its canonical representation in little-endian
+  ## byte order. Uses at most 254 bits, the remaining most-significant bits are
+  ## set to 0.
+  assert marshal(result, element.toBig(), littleEndian)
 
-func fromBytes*(_: type seq[F], bytes: openArray[byte]): seq[F] =
+iterator elements*(bytes: openArray[byte], _: type F): F =
```

Comment:
you want to call this `items` so it works with `for in` and other primitives out of the box.

Reply:
In this case, not really. I don't want every `for item in bytes` to iterate over field elements.